src/System.Linq/tests/EnumerableTests.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public partial class EnumerableTests
+    {
+        [Fact]
+        public void Average()
+        {
+            var array = Enumerable.Range(1, 10).ToArray();
+            var min = array.Min();
+            Assert.Equal(min, 1);
+        }
```

Comment:
In some cases that may be ok.  In other cases it might be overly complicated (e.g. the above example can be simply done in a single statement as ```Assert.Equals(1, Enumerable.Range(1, 10).Min());```... that's easy to read, and there's no need for the complication of lambdas here).  And for some cases it'll be insufficient (e.g. negative tests cases where you want to verify that an exception occurs).